geomstats/visualization/CategoricalDistributionsManifold.py

```diff
@@ -0,0 +1,484 @@
+from matplotlib import pyplot as plt
+from mpl_toolkits.mplot3d.art3d import Poly3DCollection
+
+
+class CategoricalDistributionsManifold:
+    r"""Class for visualizing the manifold of categorical distributions.
+
+    This is the set of $n+1$-tuples of positive reals that sum up to one,
+    i.e. the $n$-simplex. Each point is the parameter of a categorical
+    distribution, i.e. gives the probabilities of $n$ different outcomes
+    in a single experiment.
+
+    Attributes:
+    -----------
+    dim : integer
+        Dimension of the manifold.
+    points: array-like, [[..., dim + 1], [..., dim + 1], ... ]
+        Discrete points to be plotted on the manifold.
+
+    Notes:
+    ------
+    The class only implements visualization methods for 2D and 3D manifolds.
+    """
+
+    def __init__(self, dim):
+        """Construct a CategoricalDistributionsManifold object.
+
+        Construct a CategoricalDistributionsManifold with a given dimension.
+
+        Parameters:
+        -----------
+        dim : integer
+            Dimension of the manifold
+
+        Returns:
+        --------
+        None.
+
+        Notes:
+        ------
+        dim should be a positive integer.
+        The methods only support visualization of 2-D and 3-D manifolds.
+        """
+        self.dim = dim
+        self.points = []
+        self.ax = None
+        self.elev, self.azim = None, None
+        self.metric = CategoricalMetric(dim=self.dim)
+        self.dist = CategoricalDistributions(dim=self.dim)
+
+    def plot(self):
+        """Plot the 2D or 3D Manifold.
+
+        Plot the 2D Manifold as a regular 2-simplex(triangle) or
+        the 3D Manifold as a regular 3-simplex(tetrahedral).
+
+        Parameters
+        ----------
+        None.
+
+        Returns
+        -------
+        None.
+
+        Notes
+        -----
+        This method only works properly if the dimension is 2 or 3.
+
+        References
+        ----------
+        Simplex: https://en.wikipedia.org/wiki/Simplex
+        """
+        min_limit = 0
+        max_limit = 1
+        plt.figure(dpi=100)
+        self.set_axis(min_limit, max_limit)
+        if self.dim == 3:
+            self.set_view()
+            x = [0, 1, 0, 0]
+            y = [0, 0, 1, 0]
+            z = [0, 0, 0, 1]
+            vertices = [[0, 1, 2], [0, 1, 3], [0, 2, 3], [1, 2, 3]]
+            tupleList = list(zip(x, y, z))
+            poly3d = [
+                [tupleList[vertices[ix][iy]] for iy in range(len(vertices[0]))]
+                for ix in range(len(vertices))
+            ]
+            self.ax.add_collection3d(
+                Poly3DCollection(
+                    poly3d,
+                    edgecolors="k",
+                    facecolors=(0.9, 0.9, 0.9, 1.0),
+                    linewidths=3,
+                    alpha=0.2,
+                )
+            )
+
+        elif self.dim == 2:
+            X = np.linspace(start=min_limit, stop=max_limit, num=101,
```

Comment:
avoid single-letter variable names and there is no uppercase letter variable names in python except when it's class names or constant variables